rct229/rules/section23/section23rule6.py

```diff
@@ -0,0 +1,105 @@
+            list_path="$..terminals[*]",
+        )
+
+    def is_applicable(self, context, data=None):
+        rmi_b = context.baseline
+        # baseline_system_types_dict = get_baseline_system_types(rmi_b)
+        baseline_system_types_dict = {
+            "Sys-6": ["System 6"],
+            "Sys-8": [],
+        }  # This will be removed once PR #899 is merged
+
+        return any(
+            [
+                baseline_system_type_compare(system_type, applicable_sys_type, False)
+                for system_type in baseline_system_types_dict.keys()
+                for applicable_sys_type in APPLICABLE_SYS_TYPES
+            ]
+        )
+
+    def create_data(self, context, data):
+        rmi_b = context.baseline
+        # baseline_system_types_dict = get_baseline_system_types(rmi_b)
+        baseline_system_types_dict = {
+            "Sys-6": ["System 6"],
+            "Sys-8": [],
+        }  # This will be removed once PR #899 is merged
+        applicable_hvac_sys_ids = [
+            hvac_id
+            for sys_type in APPLICABLE_SYS_TYPES
+            for hvac_id in baseline_system_types_dict[sys_type]
+        ]
+
+        return {"applicable_hvac_sys_ids": applicable_hvac_sys_ids}
+
+    def list_filter(self, context_item, data):
+        applicable_hvac_sys_ids = data["applicable_hvac_sys_ids"]
+
+        return context_item.baseline["id"] in applicable_hvac_sys_ids
+
+    class TerminalRule(RuleDefinitionBase):
+        def __init__(self):
+            super(Section23Rule6.TerminalRule, self).__init__(
+                rmrs_used=UserBaselineProposedVals(False, True, False),
+                required_fields={
+                    "$": ["primary_airflow"],
+                    "fan": [
+                        "design_airflow",
+                        "design_electric_power",
+                    ],
+                },
+            )
+
+        def get_calc_vals(self, context, data=None):
+            terminal_b = context.baseline
+            design_airflow_b = terminal_b["fan"]["design_airflow"]
+            primary_airflow_b = terminal_b["primary_airflow"]
+            design_electric_power_b = terminal_b["fan"]["design_electric_power"]
+
+            return {
+                "design_airflow_b": CalcQ("air_flow_rate", design_airflow_b),
+                "primary_airflow_b": CalcQ("air_flow_rate", primary_airflow_b),
+                "design_electric_power_b": CalcQ(
+                    "electric_power", design_electric_power_b
+                ),
+            }
+
+        def rule_check(self, context, calc_vals=None, data=None):
+            design_airflow_b = calc_vals["design_airflow_b"]
+            primary_airflow_b = calc_vals["primary_airflow_b"]
+            design_electric_power_b = calc_vals["design_electric_power_b"]
+
+            return std_equal(design_airflow_b, 0.5 * primary_airflow_b) and std_equal(
+                design_electric_power_b.m, 0.35 * design_airflow_b.to("cfm").m
```

Comment:
I don't think you need to use `.m` for convert to cfm.  `std_equal()` should be able to handle the Pint Quantites directly.

Reply:
Great point! However, the reason why I put ```.m``` was because the two variables don't have the same type of units. More specifically, we compared ```electric_power``` in watts with ```air_flow``` in cfm. So, we may need to keep the code as is.

Reply:
Great point. I just updated the way you mentioned (assign ```W/cfm``` to the 0.35) because this is more elegant.